### PR TITLE
VACMS-7119: Fix broken standalone page toggle handling on submission.

### LINKED ISSUE #7119: Make CTA form validation respect the toggles for stand-alone R&S Q&A, and possibly CLP
## Describe the defect

[See slack thread from Laura Walsh](https://dsva.slack.com/archives/CDHBKAL9W/p1638380351100500)

## To Reproduce
1. Create a Q&A (or just edit https://staging.cms.va.gov/node/33830/edit)
2. Don't click the stand-alone checkbox
3. Form fails silently

## Expected behavior
Form should validate. 

Workaround: 

<details><summary>Screenshot</summary>

Edit_Q_A_How_do_I_request_an_extension_for_a_later_period_of_active_duty____VA_gov_CMS

</details>

## ACs
- [ ] remove field level requiredness and add validation of  for CTA fields only if QA is enabled a a stand-alone page.

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [x] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by ethanteague**
Resolves #7119 

## Description
See https://dsva.slack.com/archives/CDHBKAL9W/p1638380351100500
When users submit forms that have sections turned off via booleans, submissions fail when items contained in the turned off sections are required. This code was removed in https://github.com/department-of-veterans-affairs/va.gov-cms/pull/6783. This pr restores, and adds an additional check for buttons paragraphs.

## Testing done
Visual / behat.

## QA steps
 - [x] As an administrator, go to `node/37432/edit` and submit form, visually verifying that submission is successful without clicking on the standalone page bool.
 - [x] As an administrator, go to `node/add/campaign_landing_page` fill in required fields, but do not toggle on any of the optional items in the fieldsets - visually verify that form submission does not fail silently, or require items in any of the turned off toggle fields.

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -394,6 +394,46 @@ function va_gov_backend_form_alter(&$form, FormStateInterface $form_state, $form
...
+    // Mark CTA button fields as required.
+    if (!empty($form['field_buttons'])) {
+      foreach ($form['field_buttons']['widget'] as $idx => $val) {
+        if (is_numeric($idx)) {
+          $form['field_buttons']['widget'][$idx]['subform']['field_button_label']['widget'][0]['value']['#required'] = TRUE;
+          $form['field_buttons']['widget'][$idx]['subform']['field_button_link']['widget'][0]['uri']['#required'] = TRUE;
+        }
+      }
+    }
+  }
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:435]: Major:  Lots of nested loops and control structures piled into a hook_form_alter that is already 100+ lines long.    At minimum this should be put in a nice descriptive function that describes what it is doing to whom.  

Functions are easy to read and debug.  Don't be afraid to use them.  Ifs piled on loops are nightmares. 

This code also does not seem to be narrowed in scope or even describing what content types we are targeting.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -762,6 +802,29 @@ function _va_gov_backend_audience_topics_validation(array $form, FormStateInterf
...
+ */
+function _va_gov_backend_standalone_page_validation(array &$form, FormStateInterface &$form_state) {
+  if ($form_state->getValue('field_standalone_page')['value'] === 0) {
+    // If we don't unset this field, validation prevents the form from
+    // submitting. This is undesirable when the standalone option is not
+    // toggled on.
+    $form_state->unsetValue('field_related_information');
+  }
+  if (!empty($form_state->getValue('field_standalone_page')['value'])) {
+    if (empty($form_state->getValue('field_primary_category'))) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:821]:
```suggestion
  }
  elseif ((!empty($form_state->getValue('field_standalone_page')['value'])) &&
    (empty($form_state->getValue('field_primary_category')))) {
```
Minor: The second if does not need to be evaluated if 'value' was 0

**Review by swirtSJW — CHANGES_REQUESTED** [review]

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 1 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:435]: > Major:  Lots of nested loops and control structures piled into a hook_form_alter that is already 100+ lines long.    At minimum this should be put in a nice descriptive function that describes what it is doing to whom.  
> 
> 
> 
> Functions are easy to read and debug.  Don't be afraid to use them.  Ifs piled on loops are nightmares. 
> 
> 
> 
> This code also does not seem to be narrowed in scope or even describing what content types we are targeting.


Apologies, @swirtSJW - This is code Kaise wrote that was removed in 5439 and ended up breaking the bool hidden content checks, so I just restored (with an augmentation to account for the buttons paragraph) without giving refactoring much thought.

Comment (swirtSJW) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:435]: Ahhh I see.   Sorry I should have never let that in the first time around.   Can you move it  into a function and we'll call it a day?

**Review by swirtSJW — APPROVED** [review]
QA pass

**Pull request merged**